Tests/Form/Type/DocumentTypeTest.php

```diff
@@ -30,7 +30,10 @@ public function setUp()
 
     public function testDocumentManagerOptionSetsEmOption()
     {
-        $field = $this->factory->createNamed('name', 'document', null, array(
+        $typeClass = method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')
+            ? 'Doctrine\Bundle\MongoDBBundle\Form\Type\DocumentType' : 'document';
```

Comment:
So basically you can just rely on the FQCN here and remove `setName()` and `setBlockPrefix()` from the form type.